app/controllers/api/vcs.php

```diff
@@ -238,11 +243,22 @@
             $queueForBuilds
                 ->setType(BUILD_TYPE_DEPLOYMENT)
                 ->setResource($function)
-                ->setDeployment($deployment);
+                ->setDeployment($deployment)
+                ->setProject($project); // set the project because it won't be set for git deployments
+
+            $queueForBuilds->trigger(); // must trigger here so that we create a build for each function
 
             //TODO: Add event?
+        } catch (Throwable $e) {
+            $errors[] = $e->getMessage();
         }
     }
+
+    $queueForBuilds->setType(''); // prevent shutdown hook from triggering again
```

Comment:
Seems like patch. Can we introduce methods `disable` and `enable` for this purpose?

Reply:
We trigger when type is set:

https://github.com/appwrite/appwrite/blob/33b4ee5c0f41b5919ae1dac1fb86572667062f29/app/controllers/shared/api.php#L639-L653

So, I could add a disable method that sets the type to empty string. Would that suffice?

Reply:
To minimize changes, I updated reset() to properly reset all private variables (like type) and am now using that instead of setType('').